htdocs/js/jsx/behavioural_feedback_panel.jsx

```diff
@@ -0,0 +1,606 @@
+    {this.props.children}
+    </div>
+  }
+});
+
+var FeedbackPanelContent = React.createClass({
+  propTypes: {
+    feedback_level : React.PropTypes.string.isRequired,
+    feedback_values : React.PropTypes.array
+  },
+  getInitialState: function(){
+    return {
+      currentEntryToggled: null
+    }
+  },
+  markCommentToggle: function(index) {
+
+    if(index == this.state.currentEntryToggled){
+      this.setState({
+        currentEntryToggled: null
+      });
+    }
+
+    else{
+      this.setState({
+      currentEntryToggled: index
+    });
+  }
+  },
+  openThread(index){
+    this.props.open_thread(index);
+  },
+  closeThread(index){
+    this.props.close_thread(index);
+  },
+    render: function(){
+	if (this.props.feedback_level == "instrument"){
+	   var table_headers = <tr className="info"><td>Fieldname</td><td>Author</td></tr>
+	}
+	else{
+	    var table_headers = <tr className="info"><td>Author</td></tr>
+	}
+
+    if (this.props.threads.length){
+      var currentEntryToggled = this.state.currentEntryToggled;
+
+      var that = this
+      var feedbackRows = this.props.threads.map(function(row, index){
+        if (currentEntryToggled == index){
+          var thisRowCommentToggled = true;
+        }
+        else{
+          var thisRowCommentToggled = false;
+        }
+          return <FeedbackPanelRow key={row.FeedbackID} commentToggled = {thisRowCommentToggled} feedbackID={row.FeedbackID}
+	  sessionID={that.props.sessionID} type={row.Type}
+        commentID={that.props.commentID} candID={that.props.candID} status={row.QC_status} date={row.Date}
+        commentToggle={that.markCommentToggle.bind(this, index)} fieldname={row.FieldName} author={row.User}
+        onClickClose={that.props.close_thread.bind(this,index)} onClickOpen={that.props.open_thread.bind(this,index)}/>
+      }.bind(this));
+
+	    var table =
+	            <table id ="current_thread_table" className="table table-hover table-primary table-bordered dynamictable">
+        <thead id="current_thread_table_header">
+        {table_headers}
+        </thead>
+        {feedbackRows}
+        </table>
+	
+	
+      return(
+        <div className="panel-collapse collapse in">
+        <div className="panel-body">
+	  {table}
+        </div>
+        </div>
+      )
+    }
+    else{
+	return(
+	            <div className="panel-body">
+
+            There are no threads for this user!
+	            </div>
+
+      )
+    }
+  }
+});
+
+
+var FeedbackPanelRow = React.createClass({
+  getInitialState: function(){
+    return {
+      thread_entries_toggled: false,
+      thread_entries_loaded: false,
+      thread_comment_toggled: false
+    }
+  },
+  componentDidMount: function(){
+    this.loadServerState();
+  },
+  loadServerState: function(){
+    var that = this;
+
+    request = $.ajax({
+      type: "POST",
+      url: "ajax/get_thread_entry_data.php",
+      data:{
+        "callFunc1" : this.props.feedbackID
```

Comment:
This isn't a very good variable name.. and shouldn't this be a GET request, not a POST request? I'd assume get_thread_entry_data never modifies the data.